cfe_internal/enterprise/federation/federation.cf

```diff
@@ -0,0 +1,210 @@
+body file control { inputs => { "$(sys.libdir)/stdlib.cf" }; }
+
+bundle agent federation_config
+{
+  vars:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_dir" string => "/opt/cfengine/federation";
+      "bin_dir" string => "$(federation_dir)/bin";
+      "path" string => "$(federation_dir)/cfapache/federation-config.json";
+  classes:
+    enterprise_edition.(policy_server|am_policy_hub)::
+      "federation_enabled"
+        expression => fileexists("$(path)"),
+        scope => "namespace";
+  vars:
+    federation_enabled::
+      "data" data => readjson("$(path)");
+      "remotes" slist => getindices(@(data[remote_hubs]));
+  classes:
+    federation_enabled::
+      "federation_am_feeder"
+        expression => strcmp("superhub", "$(data[remote_hubs][$(remotes)][role])"),
+        scope => "namespace";
+      "federation_am_superhub"
+        expression => strcmp("feeder", "$(data[remote_hubs][$(remotes)][role])"),
+        scope => "namespace";
+      "federation_am_transporter"
+        and => {strcmp("superhub", "$(data[remote_hubs][$(remotes)][role])"),
+                strcmp("push_over_rsync", "$(data[remote_hubs][$(remotes)][transport][mode])") },
+        scope => "namespace";
+  reports:
+    federation_enabled::
```

Comment:
Should change these classes to "DEBUG_$(this.bundle).federation_enabled":: if you wnat to keep the reports